Makefile

```diff
@@ -12,21 +12,22 @@ RM      = rm -f
 MKDIR   = mkdir -p
 
 SRC  = $(wildcard src/*.c)
-COMMON_SRC = $(wildcard src/common/*.c)
-ALL_SRC = $(wildcard src/*.c src/*.h src/common/*.c src/common/*.h test/package/*.c test/cache/*.c)
+COMMON_SRC = $(wildcard src/common/*.c src/registry/*.c src/repository/*)
+ALL_SRC = $(wildcard src/*.c src/*.h src/common/*.c src/common/*.h src/registry/*.c src/registry/*.h src/repository/*.h src/repository/*.c test/package/*.c test/cache/*.c)
 SDEPS = $(wildcard deps/*/*.c)
 ODEPS = $(SDEPS:.c=.o)
 DEPS = $(filter-out $(ODEPS), $(SDEPS))
 OBJS = $(DEPS:.c=.o)
-MAKEFILES = $(wildcard deps/*/Makefile)
 
 export CC
 
+CFLAGS  += -std=gnu17 -Ideps -Isrc/common -Isrc/repository -Isrc/registry -g -Wall -Werror=return-type -Wno-unused-function $(shell curl-config --cflags)
```

Comment:
Yes, I would like that. There is currently an issue because `strdup` is both defined in `deps/strdup/strdup.c` and in `deps/url/url.h` with a `#ifdef _POSIX_C_SOURCE < 200809L` guard. This is because `strdup` is not in the c standard library. I am not sure what the best solution to this problem is. Do you know if it is possible to check if a function is already defined  to prevent multiple definitions?

Reply:
`deps/strdup/strdup.c` is already checking and providing the implementation if not available. Perhaps add its header file (?)

Reply:
But also, let me check if `std=gnu17` can cause problems...

Reply:
Well,... it seems that both compiles independently, so both will generate their `strdup` versions. Not sure if `url.c` should required `strdup.c`...

Reply:
`url.c` is not checking if someone else already defined `HAVE_STRDUP`.

Reply:
Submitted [a patch](https://github.com/jwerle/url.h/pull/8) to `url.h` to fix the `strdup` detection.

Reply:
We could add something like `#if _POSIX_C_SOURCE < 200809L && !(defined URL_H_NO_STRDUP)
` at `url.h:159`. And define `URL_H_NO_STRDUP` in `url.c`

@jwerle is there a reason that `jwelre/url.h` is header only? Otherwise we might split that into a `url.h` and `url.c` and add `clibs/strdup` as a dependency for it. That would solve our issue with `strdup`

Reply:
No real reason anymore we can totally split it up

Reply:
I've accepted the patch and I've released a new version that splits up the source

Reply:
Cool, thanks I am using the new version.